CondFormats/PPSObjects/interface/PPSAssociationCuts.h

```diff
@@ -70,6 +69,12 @@ class PPSAssociationCuts {
 
   ~PPSAssociationCuts() {}
 
+  // builds run-time data members, useful e.g. after loading data from DB
+  void initialize() {
+    for (auto &p : association_cuts_)
+      p.second.buildFunctions();
+  }
+
   const CutsPerArm &getAssociationCuts(const int sector) const { return association_cuts_.at(sector); }
```

Comment:
this is the line @malbouis meant

Reply:
Thanks! I see now. At the moment of opening the PR, there was only this answer: https://github.com/cms-sw/cmssw/issues/35936#issuecomment-956242043, which I interpret that `at` is OK for a map. Later, another answer was provided: https://github.com/cms-sw/cmssw/issues/35936#issuecomment-956255463, which seems to be in disagreement with the first one. I guess I need a decision which hint to follow. If there is no clear guideline, keeping the code as is saves my time.

Reply:
@makortel @VinInn , is it ok to keep the .at() in this case?

Reply:
Hi @jan-kaspar my interpretation is that Matti is saying `at` is OK if we are OK with throwing an exception, while Vincenzo is saying it is not OK to throw the exception, thus `at` is also not OK. I think `find` should be used.

Reply:
The real question is: does one really need a ```map```?
Is not a trivial array enough?  (what values of "arm" should be supported?)

Reply:
What would be the circumstances if an element is not found? I'm wondering mostly between `sector` comes from data (directly or via some processing) and can go out of bounds occasionally, and `sector` originally comes from configuration and out of bounds values would be logic errors.

What would the calling code do if an element is not found?

Reply:
The `sector` refers to the LHC sectors around IP5: these are two and will never change. I agree that array/vector would have been a better design. The code has been made by a student and I didn't catch this - sorry for that. Now that the class is already used in CondDB, I don't want to change it anymore. On the other hand, doing the search (by map::find) on a map with two elements is not that costly, so I believe we are having a sort of academic discussion. I will commit the change to map::find soon.

Reply:
For the moment there is no guarantee in the code that the DB payload will contain data for both LHC sectors. We assume that the person who uploads the conditions to DB does this well. Let me know if you wish me to put some "protection" to the code.